Fix #1855: Re-enabling Ktlint max-line-length check by abbreviating test function names

soamOne opened the pull request:
## Explanation
Fixes #1855: In this PR we are re-enabling the max-line-length check by abbreviating the names of test functions that violated this check.

## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

soamOne committed:
try

soamOne committed:
try

soamOne committed:
try

soamOne committed:
try

soamOne committed:
try

soamOne committed:
try

soamOne committed:
resolved in one test file

soamOne committed:
resolved in one test file

soamOne committed:
resolved in one test file

soamOne committed:
resolved in one test file

soamOne committed:
resolved in one test file

soamOne committed:
resolved in one test file

soamOne committed:
resolved in one test file

soamOne committed:
resolved in one test file

soamOne committed:
fixed lint errors

soamOne committed:
fixed lint errors

soamOne committed:
fixed lint errors

soamOne committed:
fixed lint errors

soamOne committed:
fixed lint errors

ranjsa committed:
Fix #1646:Changed all padding_# and margin_# to space_#dp in dimens value. (#1936)

* fix-dimens-value-to-follow-one-naming-convention

* fixed max line length error

* fixed space_#dpdp to space_#dp

* removed unused padding_296dp

rt4914 commented:
@soamOne Any specific reason why it is a draft PR?

soamOne commented:
> @soamOne Any specific reason why it is a draft PR?

@rt4914 I have changed just one file. I am not sure if I am abbreviating them the right way, and that is why wanted feedback.

soamOne commented:
Can you take a look at the changes @rt4914 @anandwana001 @BenHenning ?

rt4914 commented on app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityTest.kt:
```diff
@@ -99,7 +99,7 @@ class AdministratorControlsActivityTest {
   // TODO(#973): Fix AdministratorControlsActivityTest
   @Ignore
-  fun testAdministratorControlsActivity_withAdminProfile_openAdministratorControlsActivityFromNavigationDrawer_onBackPressed_showsHomeActivity() { // ktlint-disable max-line-length
+  fun testAdminControlsActivity_withAdminProfile_openAdminCtrlsActivityFromNavDrawer_HomeOnBack() {
```
Suggestion:
`testAdminCtrlActivity_withAdminProfile_openAdminControlsActivityFromNavDrawer_onBackPressed_showsHomeActivity`

If the above does not work then

`testAdminCtrl_withAdminProfile_openAdminControlsFromNavDrawer_onBackPressed_showsHome`

rt4914 commented on app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsActivityTest.kt:
```diff
@@ -283,7 +283,7 @@ class AdministratorControlsActivityTest {
 
   @Test
-  fun testAdministratorControlsFragment_loadFragment_topicUpdateOnWifiSwitchIsNotChecked_autoUpdateTopicSwitchIsNotChecked() { // ktlint-disable max-line-length
+  fun testAdminControlsFragment_OnWifiSwitchIsNotChecked_autoUpdateTopicSwitchIsNotChecked() {
```
Always make sure that the first character after `_` is lower case. 

Suggestion: 
`testAdminCtrl_topicUpdateOnWifiSwitchIsNotChecked_autoUpdateTopicSwitchIsNotChecked `

soamOne replied:
In this function @rt4914, shouldn't we have AdminCtrlFragment instead of omitting "fragment"?

rt4914 replied:
In this case/file, its file to omit the `fragment` word.

rt4914 reviewed (changes requested):
@soamOne Have a look at the suggestions.

soamOne commented:
> @soamOne Have a look at the suggestions.

Alright thanks @rt4914! working on it.

soamOne committed:
resolving ktlint maxline length

soamOne committed:
resolving ktlint maxline length

soamOne committed:
resolving ktlint maxline length

soamOne committed:
resolving ktlint maxline length

soamOne committed:
go
Merge branch 'ktlint-max-line-resolve' of https://github.com/soamOne/oppia-android into ktlint-max-line-resolve

soamOne committed:
trying

soamOne committed:
trying

soamOne committed:
trying

BenHenning commented:
I think @rt4914’s suggestions are good. Will take a look once this PR is ready for review.

rt4914 commented:
@soamOne Can you please update this PR? Merge your branch with latest develop because currently it shows 84 files changed.

BenHenning committed:
Fix #1942: Fix CI app module test flakiness/failures due to shared test processes (#1943)

* Enable full stack traces for all Robolectric tests

* Show stack traces for failing Robolectric tests.

* Introduce class-level forking & enable test parallelization.

soamOne committed:
re-enabling ktlint max line length check

soamOne committed:
re-enabling ktlint max line length check

soamOne committed:
re-enabling ktlint max line length check

soamOne committed:
re-enabling ktlint max line length check

soamOne committed:
fixing errors

soamOne committed:
fixing errors

soamOne committed:
fixing errors

soamOne committed:
lint errors fixed

soamOne committed:
lint errors fixed

prayutsu committed:
Fix #1894 : Thoroughly test FractionInputIsLessThanRuleClassifier (#1938)

* add all tests

* removed to-do

* fix suggested changes

* try fixing failing check

* change test cases order

soamOne committed:
merging

soamOne committed:
merging

soamOne committed:
Merge branch 'ktlint-max-line-resolve' of https://github.com/soamOne/oppia-android into ktlint-max-line-resolve

soamOne commented:
@rt4914 I've tried rebasing my branch but the PR is showing commits by other people as well. How do I solve it?

rt4914 commented:
> @rt4914 I've tried rebasing my branch but the PR is showing commits by other people as well. How do I solve it?

It looks like you have done this correctly but somehow github is not showing correctly. Or maybe I am missing something.
Adding @BenHenning if he has some idea here.


Meanwhile, you can do any of the following things:
1. Wait for @BenHenning 's reply and start working on another issue in parallel. https://github.com/oppia/oppia-android/issues?q=is%3Aopen+is%3Aissue+no%3Aassignee+label%3A%22SLoP+2020%22
2. Create a fresh/new PR for this same issue and close this one. 

You can do whichever seems fine to you.

@soamOne

soamOne commented:
@rt4914 I think I'll open a new PR if that is fine.

rt4914 commented:
> @rt4914 I think I'll open a new PR if that is fine.

Sure.

BenHenning commented:
@soamOne did you ensure your fork had an up-to-date develop branch when pulling down to your local checkout of it? https://github.com/oppia/oppia-android/wiki/Fork-and-Clone-Oppia-Android#updating-your-local-repository includes instructions on how to update your local checkout to include the latest main repo changes.
